WebDriverAgentTests/IntegrationTests/FBImageProcessorTests.m

```diff
@@ -9,62 +9,62 @@
 
 #import <XCTest/XCTest.h>
 
-#import "FBImageIOScaler.h"
+#import "FBImageProcessor.h"
 #import "FBIntegrationTestCase.h"
 
 
-@interface FBImageIOScalerTests : FBIntegrationTestCase
+@interface FBImageProcessorTests : FBIntegrationTestCase
 
 @property (nonatomic) NSData *originalImage;
 @property (nonatomic) CGSize originalSize;
 
 @end
 
-@implementation FBImageIOScalerTests
+@implementation FBImageProcessorTests
 
 - (void)setUp {
   XCUIApplication *app = [[XCUIApplication alloc] init];
   [app launch];
   XCUIScreenshot *screenshot = app.screenshot;
   self.originalImage = UIImageJPEGRepresentation(screenshot.image, 1.0);
-  self.originalSize = [FBImageIOScalerTests scaledSizeFromImage:screenshot.image];
+  self.originalSize = [FBImageProcessorTests scaledSizeFromImage:screenshot.image];
 }
 
 - (void)testScaling {
   CGFloat halfScale = 0.5;
-  CGSize expectedHalfScaleSize = [FBImageIOScalerTests sizeFromSize:self.originalSize scalingFactor:0.5];
+  CGSize expectedHalfScaleSize = [FBImageProcessorTests sizeFromSize:self.originalSize scalingFactor:0.5];
   [self scaleImageWithFactor:halfScale
                 expectedSize:expectedHalfScaleSize];
 
-  // 1 is the smalles scaling factor we accept
+  // 0 is the smalles scaling factor we accept
   CGFloat minScale = 0.0;
-  CGSize expectedMinScaleSize = [FBImageIOScalerTests sizeFromSize:self.originalSize scalingFactor:0.01];
+  CGSize expectedMinScaleSize = [FBImageProcessorTests sizeFromSize:self.originalSize scalingFactor:0.01];
   [self scaleImageWithFactor:minScale
                 expectedSize:expectedMinScaleSize];
 
   // For scaling factors above 100 we don't perform any scaling and just return the unmodified image
-  CGFloat unscaled = 2.0;
-  [self scaleImageWithFactor:unscaled
+  [self scaleImageWithFactor:1.0
+                expectedSize:self.originalSize];
+  [self scaleImageWithFactor:2.0
```

Comment:
Why is `scaleImageWithFactor` called twice?

Reply:
because we expect the size to not be changed if the factor is >= 1.0